domain/network/network.py

```diff
@@ -0,0 +1,171 @@
+from dataclasses import asdict, dataclass
+from enum import Enum
+from typing import List
+
+
+class NodeState(str, Enum):
+    # Node is still initializing
+    INITIALIZING = 'INITIALIZING'
+
+    # Node is ready to establish new connections, sync, and exchange transactions.
+    READY = 'READY'
+
+
+@dataclass
+class Peer:
+    """ Information about a Peer connected to a node
+
+    :param id: Peer id hash
+    :type id: str
+
+    :param app_version: The current version of app running on node
+    :type app_version: str
+
+    :param uptime: Time of node activity in seconds
+    :type uptime: float
+
+    :param address: Ip address and port of the peer
+    :type address: str
+
+    :param state: Current state of the peer
+    :type state: :py:class:`domain.network.network.NodeState`
+
+    :param last_message: Time passed since the last message that node received in seconds
+    :type last_message: float
+
+    :param latest_timestamp: Timestamp of the latest block of the node
+    :type latest_timestamp: int
+
+    :param sync_timestamp: Timestamp of the last synchronized block of the node
+    :type sync_timestamp: int
+
+    :param warning_flags: List of node warnings, if any
+    :type warning_flags: List[str]
+
+    """
+    id: str
+    app_version: str
+    uptime: float
+    address: str
+    state: NodeState
+    last_message: float
+    latest_timestamp: int
+    sync_timestamp: int
+    warning_flags: List[str]
+
+
+@dataclass
+class Network:
```

Comment:
Doesn't the name `Node` makes more sense in this class?

Reply:
Changed